Only answer if your a thief once per location

alkemann opened the pull request:
#### Summary
```SUMMARY: Interface "Only answer if you are a thief once per location grab"```

#### Purpose of change
When stealing a lot of items, you have to keep answering yes on each item. With this change the pickup loop remembers your choice for the entire location. If you say "NO I am not thief", the player only picks up all the items that is not tagged as owned.

#### Describe the solution
This can have 3 values; `ask`, `steal` or `honest` and starts out being `ask`. Upon encountering an action that would be stealing, the player is queried like before, but the answer is stored. It is reset after all pickup is done. There is also a new key binding that cycles between the 3 values.

#### Describe alternatives you've considered
I considered using enums, but the `Creature.set_value` only allows storing of strings, so then I considered constants, but ended up with just hardcoded strings.

They keybinding part can be left out and this PR is still a nice improvement.

BevapDin commented on src/pickup.cpp:
```diff
@@ -191,8 +191,15 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
-        if( !query_yn( "Picking up this item will be considered stealing, continue?" ) ) {
-            return false;
+        if( u.thief == THIEF_UNSET ) { // Has the player given input on if stealing is ok?
+            if( query_yn( "Picking up this item will be considered stealing, continue?" ) ) {
```
Needs to be translatable - see "doc/TRANSLATING.md".

alkemann replied:
Notice that I didnt add that question, but I dont mind adding _() around it. :)

davidpwbrown replied:
Yep, that one is my bad.

BevapDin commented on src/player.cpp:
```diff
@@ -514,6 +514,8 @@ player::player() :
     nv_cached = false;
     volume = 0;
 
+    thief = THIEF_UNSET; // Is player willing to steal
```
Nearly nobody is going to look into the default constructor for documentation of a member. Put documentation where you declare that member (in the header file).

kevingranade commented on src/player.h:
```diff
@@ -185,13 +191,23 @@ static const std::array< std::string, PMM_COUNT > player_movemode_str = { {
+         * HONEST: Do not pick up stolen items, even if selected in pickup
+         * STEAL: Pick up stolen items without query
+         */
+        thief_mode thief;
```
Please use creature::values for musc state like this.
https://github.com/CleverRaven/Cataclysm-DDA/blob/9119b820d5503409c180c5e5bc71db4744ebb6c9/src/creature.h#L351

alkemann replied:
Ah, thanks I will try that

alkemann commented:
Updated the description to reflect that I have moved from player prop to the Creature.set_value. Also I have added a keybinding if a player does not want to be asked once per location either.

BevapDin commented on src/pickup.cpp:
```diff
@@ -191,8 +192,39 @@ bool pick_one_up( item_location &loc, int quantity, bool &got_water, bool &offer
+                .option( "YES" ) // yes, steal all items in this location that is selected
+                .option( "NO" ) // no, pick up only what is free
+                .option( "ALWAYS" ) // Yes, steal all items and stop asking me this question
+                .option( "NEVER" ) // no, only grab free item and never ask me again
```
Have you tried this with a non-default language? The options are not defined within that context in the keybindings file and should will therefor not have any default binding, nor any translation.

alkemann replied:
I am actually trying to figure this out. If you could point me in the direction of where to create context and set bindings that would be kind. thanks.

alkemann replied:
i think I found it in keybindings.json.  But this is new text so it wont be translated anyway. Maybe I can add that in one language and try

alkemann replied:
Can you check the update @BevapDin and see if that is sufficient? I don't seem to be able to test with other languages, nor run `lang/update_pot.sh`

BevapDin replied:
Add a "name" entry to your new entries in "data/raw/keybindings.json". See the existing entries for "id": "YES".

The actual translation will be done by others, see doc/TRANSLATING.md
